test/e2e/common/node/pod_level_resources.go

```diff
@@ -0,0 +1,319 @@
+	CPULim string
+	MemReq string
+	MemLim string
+}
+
+func makeContainer(info ContainerInfo) v1.Container {
+	cmd := []string{"/bin/sh", "-c", "sleep 1d"}
+	res := getResourceInfo(info.Resources)
+	return v1.Container{
+		Name:      info.Name,
+		Command:   cmd,
+		Resources: res,
+		Image:     imageutils.GetE2EImage(imageutils.BusyBox),
+		VolumeMounts: []v1.VolumeMount{
+			{
+				Name:      "sysfscgroup",
+				MountPath: "/tmp",
+			},
+		},
+	}
+}
+
+func getResourceInfo(info *ResourceInfo) v1.ResourceRequirements {
+	var res v1.ResourceRequirements
+	if info != nil {
+		if info.CPUReq != "" || info.MemReq != "" {
+			res.Requests = make(v1.ResourceList)
+		}
+		if info.CPUReq != "" {
+			res.Requests[v1.ResourceCPU] = resource.MustParse(info.CPUReq)
+		}
+		if info.MemReq != "" {
+			res.Requests[v1.ResourceMemory] = resource.MustParse(info.MemReq)
+		}
+
+		if info.CPULim != "" || info.MemLim != "" {
+			res.Limits = make(v1.ResourceList)
+		}
+		if info.CPULim != "" {
+			res.Limits[v1.ResourceCPU] = resource.MustParse(info.CPULim)
+		}
+		if info.MemLim != "" {
+			res.Limits[v1.ResourceMemory] = resource.MustParse(info.MemLim)
+		}
+	}
+	return res
+}
+
+func makePod(metadata *metav1.ObjectMeta, podResources *ResourceInfo, containers []ContainerInfo) *v1.Pod {
+	var testContainers []v1.Container
+	for _, container := range containers {
+		testContainers = append(testContainers, makeContainer(container))
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: *metadata,
+
+		Spec: v1.PodSpec{
+			Containers: testContainers,
+			Volumes: []v1.Volume{
+				{
+					Name: "sysfscgroup",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/cgroup"},
+					},
+				},
+			},
+		},
+	}
+
+	if podResources != nil {
+		res := getResourceInfo(podResources)
+		pod.Spec.Resources = &res
+	}
+
+	return pod
+}
+
+func VerifyPodResources(gotPod v1.Pod, expectedInfo ResourceInfo) {
+	ginkgo.GinkgoHelper()
+	expectedResources := getResourceInfo(&expectedInfo)
+	gomega.Expect(gotPod.Spec.Resources).To(gomega.Equal(&expectedResources))
+}
+
+func VerifyQoS(gotPod v1.Pod, expectedQoS v1.PodQOSClass) {
+	ginkgo.GinkgoHelper()
+	gomega.Expect(gotPod.Status.QOSClass).To(gomega.Equal(expectedQoS))
+}
+
+func VerifyPodCgroups(ctx context.Context, f *framework.Framework, pod *v1.Pod, expectedResources ResourceInfo) error {
+	ginkgo.GinkgoHelper()
+	if pod.Spec.Resources != nil {
+		cmd := fmt.Sprintf("find /tmp -name '*%s*'", strings.ReplaceAll(string(pod.UID), "-", "_"))
+		framework.Logf("Namespace %s Pod %s container - finding Pod cgroup directory path: %q", cmd)
+		podCgPath, stderr, err := e2epod.ExecCommandInContainerWithFullOutput(f, pod.Name, pod.Spec.Containers[0].Name, []string{"/bin/sh", "-c", cmd}...)
+		if err != nil || len(stderr) > 0 {
+			return fmt.Errorf("encountered error while running command: %q, \nerr: %v \nstdErr: %q", cmd, err, stderr)
+		}
+
+		cpuWeightCgPath := fmt.Sprintf("%s/cpu.weight", podCgPath)
```

Comment:
Instead of hard coding these, could you define `cpu.weight` and others (e.g., `cpu.max`,...) as constants properly? Then you can join them as needed later.